src/cargo/ops/cargo_rustc/custom_build.rs

```diff
@@ -398,7 +399,8 @@ pub fn build_map<'b, 'cfg>(cx: &mut Context<'b, 'cfg>,
         // linkable to us (e.g. not a binary) and it's for the same original
         // `kind`.
         let mut ret = Vec::new();
-        for &(pkg, target, p) in cx.dep_targets(pkg, target, profile).iter() {
+        for &(pkg, target, p) in cx.dep_targets(pkg, target, Kind::Host, profile)
```

Comment:
Now that we've got `IntoIterator`, you may actually be able to omit the call to `.iter()` here.